pkg/serverless/protocol.go

```diff
@@ -81,7 +83,13 @@ type Daemon struct {
 	// work before finishing an invocation
 	InvcWg *sync.WaitGroup
 
+	// Wait on this WaitGroup to be sure that the daemon isn't shutting down
+	// to avoid double flushing in case a the flush is received between a timeout and the actual shutdown
+	TimeoutWg *sync.WaitGroup
```

Comment:
I noticed we never mark the wait group as Done. Would it make more sense to have a [mutex](https://golang.org/pkg/sync/#Mutex.Lock) in the Flush method instead? That way we can guarantee no double flushing ever occurs in any scenarios, but we also don't create a scenario where flush will totally lock off forever.